humility-core/src/hubris.rs

```diff
@@ -1834,6 +1914,33 @@ impl HubrisArchive {
                 }
 
                 match entry.tag() {
+                    gimli::constants::DW_TAG_namespace => {
+                        //
+                        // For a namespace tag, create an entry in our
+                        // namespaces vector, and push a tuple consisting
+                        // of the identifer and our depth.
+                        //
+                        let mut attrs = entry.attrs();
+
+                        while let Some(attr) = attrs.next()? {
+                            if attr.name() == gimli::constants::DW_AT_name {
+                                let name = dwarf_name(&dwarf, attr.value());
+
+                                if let Some(name) = name {
+                                    let id = self.namespaces.len();
+                                    let parent =
+                                        namespace.last().map(|(id, _)| *id);
+
+                                    self.namespaces
```

Comment:
This probably wants a struct of some kind; it's confusing to have `self.namespaces` and `namespaces`, each of which is a `Vec` containing two-item tuples with very different semantics.

It would probably be worthwhile to make types, e.g.
```rust
struct NamespacedType {
    name: String,
    parent: Option<usize>,
}
```
and
```rust
struct NamespaceEntry {
    index: usize,
    depth: usize,
}
```

Reply:
This was nagging me too; cleaned this up quite a bit.